x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPreviewDataFrameAnalyticsAction.java

```diff
@@ -101,13 +102,14 @@ protected void doExecute(Task task, Request request, ActionListener<Response> li
     }
 
     void preview(Task task, DataFrameAnalyticsConfig config, ActionListener<Response> listener) {
+        final TaskId parentTaskId = new TaskId(clusterService.getNodeName(), task.getId());
```

Comment:
I am not sure if `task.getParentTaskId()` was strictly correct. `task` is indeed the transport action task (created via our request object) that is being executed here. It SHOULD have a parent task that is the original HTTP request (I think?) but it seems that the prevailing pattern is to rely on this `task` object instead of its parent.